apps/ewallet_db/lib/ewallet_db/account.ex

```diff
@@ -262,12 +262,16 @@ defmodule EWalletDB.Account do
   @doc """
   Determine the relative depth to the master account.
 
+  Returns a non-negative integer if the account is found.
   The master account has a relative depth of 0.
+
+  It raises an error if passed nil, because a nil does not belong to any account level.
+  And it is unsafe to default it, for example, to being a top level account.
   """
-  @spec get_depth(account :: %Account{} | account_uuid :: String.t()) :: non_neg_integer()
+  @spec get_depth(%Account{} | String.t()) :: non_neg_integer() | no_return()
   def get_depth(%Account{} = account), do: get_depth(account.uuid)
 
-  def get_depth(account_uuid) do
+  def get_depth(account_uuid) when not is_nil(account_uuid) do
```

Comment:
What if I pass `nil`? Aren't we missing one more clause to raise a specific error instead of the Unmatched one that will be raised? 🤔

Reply:
Yeah actually this is intentional by me. I think `get_depth/1` should never get called or get passed directly with an external input.

So the only case `get_depth(nil)` could happen is by coding error and hence I fallback to the Elixir error.